backend/services/deployments/store/mongo/datastore_mongo.go

```diff
@@ -309,6 +312,18 @@ var (
 			Name:       &IndexArtifactProvidesName,
 		},
 	}
+
+	// 1.2.17
+	IndexDeploymentName = mongo.IndexModel{
+		Keys: bson.D{
+			{Key: StorageKeyDeploymentName,
+				Value: 1},
```

Comment:
```suggestion
			{Key: StorageKeyDeploymentName, Value: 1},
			{Key: "created", Value: 1},
```

Reply:
We also need new indexes with secondary index on `created`, possibly dropping (replacing) previous ones, to avoid collection/index scans and in-memory sorts.

Reply:
I updated the one I'm creating with additional `created` key, but I'm not sure about other ones. Looks like we only have one index with `created` field only. Should I add index with all the fields we are using in the query?